src/main/java/ch/njol/skript/expressions/ExprFilter.java

```diff
@@ -115,148 +128,61 @@ protected Object[] get(Event event) {
 		}
 	}
 
-	public Object getCurrent() {
-		return current;
-	}
-
-	private void addChild(ExprInput<?> child) {
-		children.add(child);
-	}
-
-	private void removeChild(ExprInput<?> child) {
-		children.remove(child);
-	}
-
 	@Override
 	public Class<?> getReturnType() {
-		return objects.getReturnType();
+		return unfilteredObjects.getReturnType();
 	}
 
 	@Override
 	public boolean isSingle() {
-		return objects.isSingle();
+		return false;
 	}
 
 	@Override
 	public String toString(Event event, boolean debug) {
-		return String.format("%s where [%s]", objects.toString(event, debug), rawCond);
+		return unfilteredObjects.toString(event, debug) + " that match [" + unparsedCondition + "]";
 	}
 
-	@Override
-	public boolean isLoopOf(String s) {
-		for (ExprInput<?> child : children) { // if they used player input, let's assume loop-player is valid
-			if (child.getClassInfo() == null || child.getClassInfo().getUserInputPatterns() == null)
-				continue;
+	private boolean matchesAnySpecifiedTypes(String candidateString) {
```

Comment:
i know this is my code, but I think this logic would actually be better moved to `ExprInput#isLoopOf` then `ExprFilter` can just call that

Reply:
What parts exactly would be able to be moved to ExprInput? The code within the dependent loop could be, but then the return logic would get really messy.

Reply:
I actually don't make this makes sense anymore. it would mean that ExprInput#isLoopOf could return true even though you can't actually loop it. which seems odd to me.